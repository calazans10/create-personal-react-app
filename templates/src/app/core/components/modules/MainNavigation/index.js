import React from 'react';
import { NavLink } from 'react-router-dom';
import './index.scss';

const MainNavigation = () => (
  <nav className="main-navigation">
    <ul className="main-navigation__list">
      <li className="main-navigation__item">
        <NavLink
          exact
          to="/"
          className="main-navigation__link"
          activeClassName="main-navigation__link--active"
        >
          Dashboard
        </NavLink>
      </li>
      <li className="main-navigation__item">
        <NavLink
          exact
          to="/users"
          className="main-navigation__link"
          activeClassName="main-navigation__link--active"
        >
          Usu√°rios
        </NavLink>
      </li>
    </ul>
  </nav>
);

export default MainNavigation;
