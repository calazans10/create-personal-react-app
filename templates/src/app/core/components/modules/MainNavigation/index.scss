@import 'styles/utilities';

.main-navigation {
  margin-left: rem(32);

  &__item {
    display: inline-block;

    &:not(:last-child) {
      margin-right: rem(16);
    }
  }

  &__link {
    color: #dfe4ea;
    text-decoration: none;

    &:hover {
      color: #ced6e0;
    }

    &--active {
      color: #ced6e0;
      font-weight: bold;
    }
  }
}
