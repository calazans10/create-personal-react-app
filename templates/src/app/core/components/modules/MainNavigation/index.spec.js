import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MainNavigation from './index';

describe('MainNavigation', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<MainNavigation />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
