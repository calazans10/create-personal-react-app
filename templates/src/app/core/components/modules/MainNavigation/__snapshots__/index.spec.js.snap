// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNavigation renders without crashing 1`] = `
<nav
  className="main-navigation"
>
  <ul
    className="main-navigation__list"
  >
    <li
      className="main-navigation__item"
    >
      <NavLink
        activeClassName="main-navigation__link--active"
        className="main-navigation__link"
        exact={true}
        to="/"
      >
        Dashboard
      </NavLink>
    </li>
    <li
      className="main-navigation__item"
    >
      <NavLink
        activeClassName="main-navigation__link--active"
        className="main-navigation__link"
        exact={true}
        to="/users"
      >
        Usu√°rios
      </NavLink>
    </li>
  </ul>
</nav>
`;
